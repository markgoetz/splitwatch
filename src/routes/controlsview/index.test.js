import { h } from 'preact';
import ControlsView from './index';
import { shallow } from 'preact-render-spy';

describe('ControlsView', () => {
  test('renders when playing', () => {
    const context = shallow(<ControlsView isPlaying startTime={1000} baseTime={3000} splits={[]} />);
    expect(context).toMatchSnapshot();
  });

  test('renders when paused', () => {
    const context = shallow(<ControlsView startTime={1000} baseTime={3000} splits={[]} />);
    expect(context).toMatchSnapshot();
  });

  test('renders with splits', () => {
    const context = shallow(<ControlsView startTime={1000} baseTime={3000} splits={[100, 200, 300]} />);
    expect(context).toMatchSnapshot();
  });
});