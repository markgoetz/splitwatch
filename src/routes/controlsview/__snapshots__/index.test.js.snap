// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsView renders when paused 1`] = `
preact-render-spy (1 nodes)
-------
<div>
  <Header></Header>
  <Wrapper align="center">
    <VerticalLayout align="center">
      <TimerComponent
        startTime={1000}
        baseTime={3000}
      >
      </TimerComponent>
      <SidebarLayout>
        <VerticalLayout>
          <Button
            isPrimary={true}
            onClick={[Function anonymous]}
          >
            Start
          </Button>
          <Button onClick={[Function anonymous]}>Pause</Button>
          <Button onClick={[Function anonymous]}>Stop</Button>
          <Button onClick={[Function anonymous]}>Reset</Button>
        </VerticalLayout>
        <SplitList></SplitList>
      </SidebarLayout>
    </VerticalLayout>
  </Wrapper>
</div>

`;

exports[`ControlsView renders when playing 1`] = `
preact-render-spy (1 nodes)
-------
<div>
  <Header></Header>
  <Wrapper align="center">
    <VerticalLayout align="center">
      <TimerComponent
        isPlaying={true}
        startTime={1000}
        baseTime={3000}
      >
      </TimerComponent>
      <SidebarLayout>
        <VerticalLayout>
          <Button
            isPrimary={true}
            onClick={[Function anonymous]}
          >
            Split
          </Button>
          <Button onClick={[Function anonymous]}>Pause</Button>
          <Button onClick={[Function anonymous]}>Stop</Button>
          <Button onClick={[Function anonymous]}>Reset</Button>
        </VerticalLayout>
        <SplitList></SplitList>
      </SidebarLayout>
    </VerticalLayout>
  </Wrapper>
</div>

`;
